### ISSUE TITLE
[regression] tox-gh-actions >= 3 breaks tox >= 4

### ISSUE THREAD

**Issue by alonbl** [ymyzk/tox-gh-actions#139]
# OVERVIEW

The `tox-gh-actions` somehow cause dependencies to not be installed when using recent tox.

The following output is available when running the build manually not in github, notice that the `pre-commit` which is at the `dev` dependency is not installed.

```
$ tox
.tox create: xxx/test_project/.tox/.tox
.tox installdeps: tox-gh-actions==3.*, tox >= 3.21.4
ROOT: tox-gh-actions won't override envlist because tox is not running in GitHub Actions
.pkg: install_requires> python -I -m pip install setuptools wheel
.pkg: _optional_hooks> python xxx/test_project/.tox/.tox/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_sdist> python xxx/test_project/.tox/.tox/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: install_requires_for_build_sdist> python -I -m pip install setuptools
.pkg: build_sdist> python xxx/test_project/.tox/.tox/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta
test: install_package> python -I -m pip install --force-reinstall --no-deps xxx/test_project/.tox/.tmp/package/1/test-project-0.0.0.tar.gz
test: commands[0]> pre-commit run --all-files
test: exit 2 (0.07 seconds) xxx/test_project> pre-commit run --all-files
.pkg: _exit> python xxx/test_project/.tox/.tox/lib/python3.10/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  test: FAIL code 2 (6.64=setup[6.57]+cmd[0.07] seconds)
  evaluation failed :( (6.67 seconds)
```

# REPRODUCTION

### tox.ini

```ini
[tox]
isolated_build = True
envlist = test
requires =
#    tox==3.*
#    tox-gh-actions==2.*
    tox-gh-actions==3.*

[testenv]
extras = dev
commands =
    pre-commit run --all-files {posargs}
```

### setup.cfg

```ini
[metadata]
name = test_project
long_description = file: README.md
long_description_content_type = text/markdown

[options]
packages = find_namespace:
setup_requires =
    setuptools

[options.extras_require]
dev =
    pre-commit

[egg_info]
tag_date = 0
```

### pyproject.toml

```toml
[build-system]
requires = [
    "setuptools",
    "wheel",
]
build-backend = "setuptools.build_meta"

[project]
name = "test-project"
requires-python = ">=3.10"
version = "0.0.0"
dynamic = [
    "readme",
]

[[project.authors]]
name = "Test"
email = "test@test.com"
```

### .pre-commit-config.yaml

```yaml
repos:
  - repo: https://github.com/psf/black
    rev: 22.8.0
    hooks:
      - id: black
```

### test_project/__init__.py

```python
```

## Invocation

* When `tox > 4` and `tox-gh-actions==3.*` fails.
* When `tox > 4` and `tox-gh-action==2.*` succeeds.
* when `tox==3.*` and `tox-gh-actions==2.*` success.

Please play with the dependencies at `tox.ini`:

```
requires =
#    tox==3.*
#    tox-gh-actions==2.*
    tox-gh-actions==3.*
```

# SUSPECTED CAUSE

The `tox-gh-actions` causes requirements of environment to be skipped when `tox >= 4`

# EXPECTED BEHAVIOR

* `tox-gh-actions` should not affect build environment when running out of github
* `tox-gh-actions` should not fail with recent `tox`

**Comment by ymyzk** [comment]
Hello @alonbl, thanks for filing an issue. Let me share the result of my initial investigation.

I tried to reproduce the issue on my machine using the given example and I found that we get the same error even when tox-gh-actions is **not** installed.

At this point, I feel this is not an issue with tox-gh-actions. This may be rather an issue with tox 3 -> 4 upgrade or your tox   configuration is not compatible with tox v4.
